### ISSUE TITLE
`class-methods-use-this` doesn't catch when this is used in a function defined within the scope of the method

### ISSUE THREAD

**Issue by andrewdtalley-3** [eslint/eslint#13527]
**Tell us about your environment**

* **ESLint Version:** 7.5.0
* **Node Version:** 12.16.1
* **npm Version:** 6.14.5

**What parser (default, Babel-ESLint, etc.) are you using?**
Default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
    "env": {
        "es2020": true,
        "node": true
    },
    "extends": "eslint:recommended",
    "parserOptions": {
        "ecmaVersion": 11,
        "sourceType": "module"
    },
    "rules": {
        "class-methods-use-this": "error"
    }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

This is a minimal reproduction of the issue, not the actual code I first ran into the issue with.

```js
// index.js
class MyClass {
  constructor(attr) {
    this.attr = attr;
  }

  someMethod() {
    function subFn() {
      console.log(this.attr);
    }

    subFn();
  }
}
```

```bash
npx eslint index.js
```

**What did you expect to happen?**
I would expect this not to conflict with the `class-methods-use-this` rule – the function `subFn` is using this, and is defined within the method, so `someMethod` is using `this` and couldn't be moved outside of the class.

**What actually happened? Please include the actual, raw output from ESLint.**
It says this violates the `class-methods-use-this` rule.

```bash
  1:7   error  'MyClass' is defined but never used                      no-unused-vars
  6:13  error  Expected 'this' to be used by class method 'someMethod'  class-methods-use-this
```

**Are you willing to submit a pull request to fix this bug?**
No.

**Comment by andrewdtalley-3** [comment]
Sorry, realized this is invalid code – closing the issue.
